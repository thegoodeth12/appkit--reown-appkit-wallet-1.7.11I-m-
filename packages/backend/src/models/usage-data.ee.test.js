import { describe, it, expect } from 'vitest';
import UsageData from '@/models/usage-data.ee.js';
import User from '@/models/user.js';
import Subscription from '@/models/subscription.ee.js';
import Base from '@/models/base.js';
import { createUsageData } from '@/factories/usage-data.js';

describe('UsageData model', () => {
  it('tableName should return correct name', () => {
    expect(UsageData.tableName).toBe('usage_data');
  });

  it('jsonSchema should have correct validations', () => {
    expect(UsageData.jsonSchema).toMatchSnapshot();
  });

  it('relationMappings should return correct associations', () => {
    const relationMappings = UsageData.relationMappings();

    const expectedRelations = {
      user: {
        relation: Base.BelongsToOneRelation,
        modelClass: User,
        join: {
          from: 'usage_data.user_id',
          to: 'users.id',
        },
      },
      subscription: {
        relation: Base.BelongsToOneRelation,
        modelClass: Subscription,
        join: {
          from: 'usage_data.subscription_id',
          to: 'subscriptions.id',
        },
      },
    };

    expect(relationMappings).toStrictEqual(expectedRelations);
  });

  it('increaseConsumedTaskCountByOne should increase consumed task count by one', async () => {
    const usageData = await createUsageData({
      consumedTaskCount: 1234,
    });

    await usageData.increaseConsumedTaskCountByOne();
    const refetchedUsageData = await usageData.$query();

    expect(refetchedUsageData.consumedTaskCount).toStrictEqual(1235);
  });
});
