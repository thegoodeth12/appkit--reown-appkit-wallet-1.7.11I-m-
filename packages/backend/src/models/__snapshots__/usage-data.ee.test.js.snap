// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UsageData model > jsonSchema should have correct validations 1`] = `
{
  "properties": {
    "consumedTaskCount": {
      "type": "integer",
    },
    "createdAt": {
      "type": "string",
    },
    "deletedAt": {
      "type": "string",
    },
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "nextResetAt": {
      "type": "string",
    },
    "subscriptionId": {
      "format": "uuid",
      "type": "string",
    },
    "updatedAt": {
      "type": "string",
    },
    "userId": {
      "format": "uuid",
      "type": "string",
    },
  },
  "required": [
    "userId",
    "consumedTaskCount",
    "nextResetAt",
  ],
  "type": "object",
}
`;
